server/src/structs.rs

```diff
@@ -0,0 +1,231 @@
+    /// Get the asset's delegatee. If the asset is not delegated, this
+    /// will return None.
+    fn delegatee(&self) -> Option<Uuid>;
+    /// Get the asset's creation time.
+    fn created_at(&self) -> SystemTime;
+    /// Get the asset's last update time. If the asset has never been
+    /// updated, this will return None.
+    fn updated_at(&self) -> Option<SystemTime>;
+    /// Check if the asset is grouped.
+    fn is_grouped(&self) -> bool;
+    /// Check if the asset is delegated.
+    fn is_delegated(&self) -> bool;
+    /// Check if the asset is open to the public.
+    fn is_public(&self) -> bool;
+    /// Get the list of clients that have access to the asset.
+    fn clients(&self) -> Vec<Uuid>;
+    /// Get the status of the asset.
+    fn status(&self) -> AssetStatus;
+}
+
+#[duplicate_item(asset; [Aircraft]; [Vertiport])]
+impl AssetsInfo for asset {
+    fn id(&self) -> Uuid {
+        self.id
+    }
+    fn group_id(&self) -> Option<Uuid> {
+        self.group_id
+    }
+    fn name(&self) -> String {
+        self.full_name()
+    }
+    fn owner(&self) -> Uuid {
+        self.owner
+    }
+    fn delegatee(&self) -> Option<Uuid> {
+        self.delegatee
+    }
+    fn created_at(&self) -> SystemTime {
+        self.created_at
+    }
+    fn updated_at(&self) -> Option<SystemTime> {
+        self.updated_at
+    }
+    fn is_grouped(&self) -> bool {
+        self.group_id.is_some()
+    }
+    fn is_delegated(&self) -> bool {
+        self.delegatee.is_some()
+    }
+    fn is_public(&self) -> bool {
+        self.restricted_to.is_empty()
+    }
+    fn clients(&self) -> Vec<Uuid> {
+        self.restricted_to.clone()
+    }
+    fn status(&self) -> AssetStatus {
+        self.status.clone()
+    }
+}
+
+/// Status of an asset.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum AssetStatus {
+    /// The asset is available for use.
+    Available,
+    /// The asset is unavailable for use.
+    Unavailable,
+    /// The asset is only available for emergencies.
+    Emergency,
+}
+
+/// A struct representing a location.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Location {
+    pub latitude: OrderedFloat<f64>,
+    pub longitude: OrderedFloat<f64>,
+}
+
+// =====================================================================
+// Asset types
+// =====================================================================
+
+/// A struct representing an aircraft.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Aircraft {
+    pub id: Uuid,
+    pub name: Option<String>,
+    pub group_id: Option<Uuid>,
+    pub owner: Uuid,
+    pub delegatee: Option<Uuid>,
+    pub created_at: SystemTime,
+    pub updated_at: Option<SystemTime>,
+    /// If the vector is empty, the aircraft is available to everyone.
+    /// Otherwise, the aircraft is only available to the clients in the
+    /// vector.
+    pub restricted_to: Vec<Uuid>,
```

Comment:
is `whitelist` fine?

noted these are temporary

Reply:
[21cddfd](https://github.com/Arrow-air/svc-assets/pull/2/commits/21cddfde26697af3ef9860ef25210cb3acff23e2)